spring-kafka/src/main/java/org/springframework/kafka/support/converter/BatchMessageConverter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support.converter;
+
+import java.lang.reflect.Type;
+import java.util.List;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import org.springframework.kafka.support.Acknowledgment;
+import org.springframework.messaging.Message;
+
+/**
+ * A Kafka-specific {@link Message} converter strategy.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Gary Russell
+ */
+public interface BatchMessageConverter<K, V> extends MessageConverter {
```

Comment:
I couldn't make it so - some limitation with generics and containers.

When I made it non-generic, for

```java
	@SuppressWarnings("unchecked")
	protected Message<?> toMessagingMessage(List<ConsumerRecord<K, V>> records, Acknowledgment acknowledgment) {
		return getBatchMessageConverter().toMessage(records, acknowledgment, this.inferredType);
	}
```

I get 

    Description	Resource	Path	Location	Type
    The method toMessage(List<ConsumerRecord<?,?>>, Acknowledgment, Type) in the type BatchMessageConverter is not applicable for the arguments (List<ConsumerRecord<K,V>>, Acknowledgment, Type)	BatchMessagingMessageListenerAdapter.java	/spring-kafka/src/main/java/org/springframework/kafka/listener/adapter	line 124	Java Problem

I'd be happy if you know a work-around other than making it generic.